src/test/javascript/spec/component/admin/health.component.spec.ts

```diff
@@ -6,61 +6,86 @@ import { ArtemisTestModule } from '../../test.module';
 import { HealthComponent } from 'app/admin/health/health.component';
 import { HealthService } from 'app/admin/health/health.service';
 import { Health } from 'app/admin/health/health.model';
+import { By } from '@angular/platform-browser';
+import { HealthModalComponent } from 'app/admin/health/health-modal.component';
+import { MockComponent, MockDirective } from 'ng-mocks';
+import { MockNgbModalService } from '../../helpers/mocks/service/mock-ngb-modal.service';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { TranslatePipeMock } from '../../helpers/mocks/service/mock-translate.service';
+import { JhiConnectionStatusComponent } from 'app/shared/connection-status/connection-status.component';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
 
 describe('HealthComponent', () => {
     let comp: HealthComponent;
     let fixture: ComponentFixture<HealthComponent>;
-    let service: HealthService;
+    let healthService: HealthService;
+    let modalService: NgbModal;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
             imports: [ArtemisTestModule],
-            declarations: [HealthComponent],
-        })
-            .overrideTemplate(HealthComponent, '')
-            .compileComponents();
+            declarations: [HealthComponent, MockComponent(HealthModalComponent), TranslatePipeMock, MockComponent(JhiConnectionStatusComponent), MockDirective(TranslateDirective)],
+            providers: [{ provide: NgbModal, useClass: MockNgbModalService }],
+        }).compileComponents();
     });
 
     beforeEach(() => {
         fixture = TestBed.createComponent(HealthComponent);
         comp = fixture.componentInstance;
-        service = fixture.debugElement.injector.get(HealthService);
+        healthService = TestBed.inject(HealthService);
+        modalService = TestBed.inject(NgbModal);
```

Comment:
Could we move this in a `then` branch and remove the second `beforeEach()` statement?

Reply:
added in e94431d